test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1

```diff
@@ -2,6 +2,33 @@
 # Licensed under the MIT License.
 [System.Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidGlobalVars", "")]
 param()
+
+Describe "Behavior is specific for each platform" {
+    BeforeAll {
+        $skipTests = $false
+        if ($EnabledExperimentalFeatures -notcontains 'PSNativeCommandArgumentPassing') {
+            $skipTests = $true
+        }
+    }
+    It "PSNativeCommandArgumentPassing is set to 'Windows' on Windows systems" -skip:(-not $IsWindows) {
+        $PSNativeCommandArgumentPassing | Should -Be "Windows"
+    }
+    It "PSNativeCommandArgumentPassing is set to 'Standard' on non-Windows systems" -skip:($IsWindows) {
+        $PSNativeCommandArgumentPassing | Should -Be "Standard"
+    }
+    It "Has proper behavior on Windows" -skip:(-not $IsWindows) {
+        "@echo off`nSET V1=1" > "$TESTDRIVE\script 1.cmd"
```

Comment:
for batch file case, you can have `-testcases` for extension `cmd` and `bat`.  should have tests for .wsf, .vbs, and .js.  You'll probably want to do this before running those tests to set cscript as default (as wscript is the default otherwise), need to run elevated:

```
cscript //h:cscript //nologo //s
```

test.wsf

```xml
<?xml version="1.0" ?>
 <job id="test">
   <script language="VBScript">
     <![CDATA[
for i = 0 to wScript.arguments.count-1
    wscript.echo "Argument " & i & " is: " & wScript.arguments(i)
next
     ]]>
   </script>
 </job>
```

test.vbs

```vbscript
for i = 0 to wScript.arguments.count-1
    wscript.echo "Argument " & i & " is: " & wScript.arguments(i)
next
```

test.js

```javascript
for (i = 0; i < WScript.Arguments.Count(); i++) {
    WScript.echo("Argument " + i + " is: " + WScript.Arguments(i));
}
```

Reply:
@SteveL-MSFT how do you get the `test.js` to run? I can't get it to go without a prepended `cscript`

Reply:
`.js` should be associated with wscript by default (hence the cscript command above)